dali/operators/image/remap/cvcuda/warp_perspective.cc

```diff
@@ -84,6 +84,62 @@ the top-left pixel (like in OpenCV).))doc", "corner")
                           "Otherwise it's interpreted as source to destination "
                           "coordinates mapping.", true);
 
+bool OCVCompatArg(const std::string &arg) {
+  if (arg == "corner") {
+    return false;
+  } else if (arg == "center") {
+    return true;
+  } else {
+    DALI_FAIL(make_string("Invalid pixel_origin argument: ", arg));
+  }
+}
+
+template<typename T>
+T GetFillValue(const std::vector<float>& fill_value_arg, int channels) {
+  if (fill_value_arg.size() > 1) {
+    if (channels > 0) {
+      if (channels != static_cast<int>(fill_value_arg.size())) {
+        DALI_FAIL(make_string(
+            "Number of values provided as a fill_value should match the number of channels.\n"
+            "Number of channels: ",
+            channels, ". Number of values provided: ", fill_value_arg.size(), "."));
+      }
+      assert(channels <= 4);
+      T fill_value{0, 0, 0, 0};
+      if constexpr (std::is_same<T, float4>::value) {
+        std::memcpy(&fill_value, fill_value_arg.data(), fill_value_arg.size() * sizeof(float));
+      } else if constexpr (std::is_same<T, cv::Scalar>::value) {
+        std::copy(fill_value_arg.begin(), fill_value_arg.end(), fill_value.val);
+      } else {
+        static_assert(false, "Unsupported type");
```

Comment:
This won't compile.

Reply:
I forgot some compilers don't like this, I only tested gcc-13. There's usually a trick around this shown below, shall I do this, or a throw a runtime error, or just don't worry throwing errors since this is a niche function private to warp_perspective?

```c++
template<typename T> struct always_false : std::false_type {};
template<typename T> constexpr bool always_false_v = always_false<T>::value;
constexpr if (...) {
  ...
} else {
   static_assert(detail::always_false_v<T>, "error message");
}
```